### PR TITLE
added poetry for simplifying building and running the package

### PR DESCRIPTION
**Pull request by Seamooo**
This PR is in relation to issue #3 
There are quite a few changes in this PR
[poetry](https://python-poetry.org/) has been added for management of dependencies, packaging, and building
imports have been made to depend off the module `mdtohtml`
both the resources and assets have been moved inside the module file such that they may be accessed with relative imports
Small fixes have been added such that it's linux compatible.

Required Changes:
the pyproject.toml file will need to be filled out with author information, and a description of the project.
Additionally if there is a better name for the module this should be decided upon likely in this PR.

Nice to haves:
potentially the README should be populated with a developer section about poetry
ie that to run the application now it requires running `poetry install` then `poetry run python3 main.py`

Note: very little testing has been done, however the changes made have been tested

### CODE REVIEW

**Review by berkerdemoglu — COMMENTED** [review]
`poetry.lock`, `pyproject.toml` and tests package initializer are all fine.

**Review by berkerdemoglu — CHANGES_REQUESTED** [review]
However, the project's name is *md-2-html*, not *mdtohtml* so please change that. Another thing is, have you run the code? Does it produce the same result? What command did you use to run it?

**Comment by berkerdemoglu** [comment]
Another comment, we should add actions to check if the code runs properly. create a new directory for the GUI as it looks as though it will need several files. put lexer in its own package

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[tool.poetry]
+name = "md-2-html"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (Seamooo) [inline pyproject.toml:5]: @berkerdemoglu very minor but filling this out before deploying as a pypi package (if that's what you envision), is required.

**Comment by Seamooo** [comment]
> However, the project's name is _md-2-html_, not _mdtohtml_ so please change that. Another thing is, have you run the code? Does it produce the same result? What command did you use to run it?

I mentioned how to run this in the original PR comment, but will expand in slightly more detail. Navigating to the repository (in root or subdirectory) run `poetry install`. This will install the dependencies of this package, and egg reference to the project in a cached venv. Navigating to `md_2_html` run `poetry run python main.py` to start the gui.

**Comment by Seamooo** [comment]
> Another comment, we should add actions to check if the code runs properly. create a new directory for the GUI as it looks as though it will need several files. put lexer in its own package

Specifically for adding actions to check the code runs properly this should likely be done in a seperate PR as some additional specification is required for what's expected in terms of python-version / os compatibility

File: pyproject.toml (hunk 1 above)
Comment (berkerdemoglu) [reply pyproject.toml:5]: sure, i'll replace `Your Name <you@example.com>` with my name and my email after the PR. i will open an issue on deploying as a PyPI package.

**Review by berkerdemoglu — APPROVED** [review]

**Comment by Seamooo** [comment]
ready to merge (I can't)

**Comment by berkerdemoglu** [comment]
> ready to merge (I can't)

yes im just testing the code and making sure that everything works and i understand what's going on. will probably merge the PR tonight.

**Pull request merged**